--- brew-install-1.10.3.986/deps.edn.orig	2021-10-03 01:12:35.324493393 +0200
+++ brew-install-1.10.3.986/deps.edn	2021-10-03 01:14:02.199311312 +0200
@@ -7,6 +7,6 @@
 
  :aliases
  {:build    ;; clj -T:build release
-  {:deps {io.github.clojure/tools.build {:git/tag "v0.5.0" :git/sha "7d77952"}}
+  {:deps {io.github.clojure/tools.build {:local/root "../tools.build-0.5.1"}}
    :paths ["script"]
    :ns-default build}}}
