From ca0734a9f355fdab8411637cc2391c0330239aac Mon Sep 17 00:00:00 2001
From: Dennis Schridde <devurandom@gmx.net>
Date: Fri, 4 Oct 2019 15:37:33 +0200
Subject: [PATCH] onig_sys/build.rs: Error if oniguruma.h cannot be found

---
 onig_sys/build.rs | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/onig_sys/build.rs b/onig_sys/build.rs
index e91d038..779b7cc 100644
--- a/onig_sys/build.rs
+++ b/onig_sys/build.rs
@@ -210,14 +210,16 @@ pub fn main() {
         }
         match conf.probe("oniguruma") {
             Ok(lib) => {
-                for path in lib.include_paths {
+                for path in &lib.include_paths {
                     let header = path.join("oniguruma.h");
                     if header.exists() {
                         bindgen_headers(&header.display().to_string());
-                        break;
+                        return
                     }
                 }
-                return
+                if require_pkg_config {
+                    panic!("Unable to find oniguruma.h in include paths from pkg-config: {:?}", lib.include_paths);
+                }
             },
             Err(ref err) if require_pkg_config => {
                 panic!("Unable to find oniguruma in pkg-config, and RUSTONIG_SYSTEM_LIBONIG is set: {}", err);
-- 
2.23.0

