--- blender-2.79b-orig/intern/openvdb/intern/openvdb_writer.cc	2018-03-15 22:26:32.000000000 +1100
+++ blender-2.79b/intern/openvdb/intern/openvdb_writer.cc	2018-05-30 21:23:46.006974503 +1000
@@ -45,7 +45,7 @@
 
 void OpenVDBWriter::insert(const openvdb::GridBase &grid)
 {
-#if (OPENVDB_LIBRARY_MAJOR_VERSION_NUMBER == 3)
+#if (OPENVDB_LIBRARY_MAJOR_VERSION_NUMBER >= 3)
 	m_grids->push_back(grid.copyGrid());
 #else
 	m_grids->push_back(grid.copyGridWithNewTree());
