Index: gdal/ogr/ogrsf_frmts/mysql/ogr_mysql.h
===================================================================
--- gdal/ogr/ogrsf_frmts/mysql/ogr_mysql.h	(revision 40362)
+++ gdal/ogr/ogrsf_frmts/mysql/ogr_mysql.h	(revision 40363)
@@ -44,5 +44,4 @@
 #endif
 
-#include <my_global.h>
 #include <mysql.h>
 
Index: gdal/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp
===================================================================
--- gdal/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp	(revision 40362)
+++ gdal/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp	(revision 40363)
@@ -32,28 +32,4 @@
 #include "ogr_mysql.h"
 
-#ifdef _MSC_VER
-#pragma warning( push )
-#pragma warning( disable : 4201 ) /* nonstandard extension used : nameless struct/union */
-#endif
-#if defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 8))
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wzero-as-null-pointer-constant"
-#endif
-#ifdef __clang__
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wunknown-pragmas"
-#pragma clang diagnostic ignored "-Wdocumentation"
-#endif
-#include <my_sys.h>
-#ifdef __clang__
-#pragma clang diagnostic pop
-#endif
-#if defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 8))
-#pragma GCC diagnostic pop
-#endif
-#ifdef _MSC_VER
-#pragma warning( pop )
-#endif
-
 #include "cpl_conv.h"
 #include "cpl_string.h"
