From 124f0343436d1267319ac627fc220530091b41ea Mon Sep 17 00:00:00 2001
From: Even Rouault <even.rouault@spatialys.com>
Date: Sat, 5 May 2018 22:18:37 +0200
Subject: [PATCH] Fix build with recent Poppler (0.64) (fix #573)

---
 frmts/pdf/pdfobject.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/frmts/pdf/pdfobject.cpp b/frmts/pdf/pdfobject.cpp
index ae734a74e4c..5eab5dd1546 100644
--- a/frmts/pdf/pdfobject.cpp
+++ b/frmts/pdf/pdfobject.cpp
@@ -1055,7 +1055,12 @@ const CPLString& GDALPDFObjectPoppler::GetString()
 {
     if (GetType() == PDFObjectType_String)
     {
+#ifdef POPPLER_0_58_OR_LATER
+        // At least available since poppler 0.41
+        const GooString* gooString = m_po->getString();
+#else
         GooString* gooString = m_po->getString();
+#endif
         return (osStr = GDALPDFGetUTF8StringFromBytes(reinterpret_cast<const GByte*>(gooString->getCString()),
                                                       static_cast<int>(gooString->getLength())));
     }
