--- a/llvm-lua/CMakeLists.txt
+++ b/llvm-lua/CMakeLists.txt
@@ -1,4 +1,3 @@
-
 #
 # llvm-lua source
 #
@@ -83,20 +82,23 @@
 # Lua library.
 #
 set(LLVM_LUA_LIBS)
+
 # llvm-lua static library.
 add_library(llvm-lua_static STATIC ${LLVM_LUA_LIB_SRC})
 add_target_properties(llvm-lua_static COMPILE_FLAGS "${COMMON_CFLAGS} ${LLVM_CFLAGS}")
 add_target_properties(llvm-lua_static LINK_FLAGS "${LD_FLAGS} ${COMMON_LDFLAGS} ${LLVM_LFLAGS}")
 target_link_libraries(llvm-lua_static ${LIBS} ${LLVM_ALL_LIBS})
+
 set(LLVM_LUA_MAIN_LIB llvm-lua_static)
+
 if(WANT_STATIC_LIBRARY)
 	# lua static library without LLVM JIT support.
-	add_library(lua_static STATIC ${NO_JIT_SRC})
-	add_target_properties(lua_static COMPILE_FLAGS "${COMMON_CFLAGS}")
-	add_target_properties(lua_static LINK_FLAGS "${LD_FLAGS} ${COMMON_LDFLAGS}")
-	target_link_libraries(lua_static ${LIBS})
-
-	set(LLVM_LUA_LIBS llvm-lua_static lua_static)
+	add_library(llvm-lua_nojit_static STATIC ${NO_JIT_SRC})
+	add_target_properties(llvm-lua_nojit_static COMPILE_FLAGS "${COMMON_CFLAGS}")
+	add_target_properties(llvm-lua_nojit_static LINK_FLAGS "${LD_FLAGS} ${COMMON_LDFLAGS}")
+	target_link_libraries(llvm-lua_nojit_static ${LIBS})
+
+	set(LLVM_LUA_LIBS llvm-lua_static llvm-lua_nojit_static)
 endif(WANT_STATIC_LIBRARY)
 
 if(WANT_SHARED_LIBRARY)
@@ -108,15 +110,15 @@
 	set_target_properties(llvm-lua_dynamic PROPERTIES OUTPUT_NAME llvm-lua)
 
 	# lua dynamic library without LLVM JIT support.
-	add_library(lua_dynamic SHARED ${NO_JIT_SRC})
-	add_target_properties(lua_dynamic COMPILE_FLAGS "${COMMON_CFLAGS}")
-	add_target_properties(lua_dynamic LINK_FLAGS "${LD_FLAGS} ${COMMON_LDFLAGS}")
-	target_link_libraries(lua_dynamic ${LIBS})
-	set_target_properties(lua_dynamic PROPERTIES OUTPUT_NAME lua)
+	add_library(llvm-lua_nojit_dynamic SHARED ${NO_JIT_SRC})
+	add_target_properties(llvm-lua_nojit_dynamic COMPILE_FLAGS "${COMMON_CFLAGS}")
+	add_target_properties(llvm-lua_nojit_dynamic LINK_FLAGS "${LD_FLAGS} ${COMMON_LDFLAGS}")
+	target_link_libraries(llvm-lua_nojit_dynamic ${LIBS})
+	set_target_properties(llvm-lua_nojit_dynamic PROPERTIES OUTPUT_NAME llvm-lua_nojit)
 
 	# link lua interpreter with dynamic library
 	set(LLVM_LUA_MAIN_LIB llvm-lua_dynamic)
-	set(LLVM_LUA_LIBS ${LLVM_LUA_LIBS} llvm-lua_dynamic lua_dynamic)
+	set(LLVM_LUA_LIBS ${LLVM_LUA_LIBS} llvm-lua_dynamic llvm-lua_nojit_dynamic)
 endif(WANT_SHARED_LIBRARY)
 
 #
@@ -303,4 +305,3 @@
 	endif(LLVM_LUA_LIBS)
 
 endif(APPLE AND WANT_SHARED_LIBRARY AND WANT_FRAMEWORK)
-
