Description: default.prefs file is installed in /etc/dspam
Author: Debian DSPAM Maintainers <pkg-dspam-misc@lists.alioth.debian.org>
Forwarded: not-needed

--- a/webui/cgi-bin/admin.cgi
+++ b/webui/cgi-bin/admin.cgi
@@ -167,14 +167,6 @@
 
   if ($FORM{'username'} eq "" || $FORM{'username'} eq "default") {
     $FILE = "/etc/dspam/default.prefs";
-    if ($CONFIG{'PREFERENCES_EXTENSION'} != 1 && ! -l "/etc/dspam/default.prefs") {
-      $DATA{'ERROR'} = "<em>WARNING:</em> " .
-        "These default preferences will not be loaded by DSPAM, but only by ".
-        " the CGI interface when a user initially sets up their preferences. ".
-        "To have DSPAM override its configuration with these default ".
-        "preferences, symlink $CONFIG{'DSPAM_HOME'}/default.prefs to the ".
-        "default.prefs file in the CGI directory.<BR><BR>";
-    }
   } else {
     $FILE = GetPath($FORM{'username'}) . ".prefs";
   }
--- a/src/pref.c
+++ b/src/pref.c
@@ -180,7 +180,7 @@
   PTX[0] = NULL;
 
   if (user == NULL) {
-    snprintf(filename, MAX_FILENAME_LENGTH, "%s/default.prefs", home);
+    snprintf(filename, MAX_FILENAME_LENGTH, "/etc/dspam/default.prefs", home);
   } else {
     _ds_userdir_path (filename, home, user, "prefs");
   }
@@ -310,7 +310,7 @@
   ignore = ignore; /* Keep compiler happy */
 
   if (username == NULL) {
-    snprintf(filename, MAX_FILENAME_LENGTH, "%s/default.prefs", home);
+    snprintf(filename, MAX_FILENAME_LENGTH, "/etc/dspam/default.prefs", home);
   } else {
     _ds_userdir_path (filename, home, username, "prefs");
   }
@@ -339,7 +339,7 @@
   ignore = ignore; /* Keep compiler happy */
 
   if (username == NULL) {
-    snprintf(filename, MAX_FILENAME_LENGTH, "%s/default.prefs", home);
+    snprintf(filename, MAX_FILENAME_LENGTH, "/etc/dspam/default.prefs", home);
   } else {
     _ds_userdir_path (filename, home, username, "prefs");
   }
