--- gtk+-2.24.31/gtk/gtk.symbols.orig	2018-03-18 15:17:39.296719190 -0000
+++ gtk+-2.24.31/gtk/gtk.symbols	2018-03-18 15:17:57.490053094 -0000
@@ -2498,8 +2498,8 @@
 #endif
 #endif
 
-#if IN_HEADER(__gtk_marshal_MARSHAL_H__)
-#if IN_FILE(__gtk_marshal_MARSHAL_C__)
+#if IN_HEADER(__gtk_marshal_MARSHAL_H__) || IN_HEADER(__GTK_MARSHAL_MARSHAL_H__)
+#if IN_FILE(__gtk_marshal_MARSHAL_C__) || IN_FILE(__GTK_MARSHAL_MARSHAL_C__)
 #ifndef GTK_DISABLE_DEPRECATED
 gtk_marshal_BOOLEAN__POINTER
 gtk_marshal_BOOLEAN__POINTER_INT_INT
